### PR TITLE
[SD-2740],[SD-2753] Simple package publishing to the public API

### PR DESCRIPTION
**Pull request by marwoodandrew**
Publish a simple package of an article and a picture to the API.

There is a known issue in that the package reference is to the actual text article, it should be to the digital package.

### CODE REVIEW

File: server/apps/publish/archive_publish.py [hunk 1]
```diff
@@ -124,19 +124,22 @@ def update(self, id, updates, original):
...
-                            updated = copy(original)
-                            updated.update(updates)
-                            # create a takes package
-                            package_id = TakesPackageService().package_story_as_a_take(updated, {}, None)
-                            original = get_resource_service('archive').find_one(req=None, _id=original['_id'])
-                            queued_digital, takes_package = self._publish_takes_package(package_id, updates,
-                                                                                        original, last_updated)
+                        # if text or preformatted item is going to be sent to digital subscribers, package it as a take
+                        if self.sending_to_digital_subscribers(original):
+                            if (original['type'] == 'text' or original['type'] == 'preformatted'):
```
Comment (sivakuna-aap) [inline server/apps/publish/archive_publish.py]: Please use constants which @mdhaman has put for content type.

File: server/apps/publish/archive_publish.py [hunk 2]
```diff
@@ -124,19 +124,23 @@ def update(self, id, updates, original):
...
-                            updated.update(updates)
-                            # create a takes package
-                            package_id = TakesPackageService().package_story_as_a_take(updated, {}, None)
-                            original = get_resource_service('archive').find_one(req=None, _id=original['_id'])
-                            queued_digital, takes_package = self._publish_takes_package(package_id, updates,
-                                                                                        original, last_updated)
+                        # if text or preformatted item is going to be sent to digital subscribers, package it as a take
+                        if self.sending_to_digital_subscribers(original):
+                            # takes packages are only created for these types
+                            if (original['type'] == CONTENT_TYPE.TEXT or original['type'] == CONTENT_TYPE.PREFORMATTED):
```
Comment (plamut) [inline server/apps/publish/archive_publish.py]: If you feel this line is a bit lengthy, you can shorten it to the following (esp. useful if there are 3 or more checks):
```py
original['type'] in (CONTENT_TYPE.TEXT, CONTENT_TYPE.PREFORMATTED)
```
No need to change it in this particular case though, just a tip.

**Comment by plamut** [comment]
Looks good so far IMO, apart from a few failed tests that need to be fixed.

**Comment by ioanpocol** [comment]
:+1:  (except the 2 failing behave tests)

**Comment by mugurrus** [comment]
:+1: (just fix tests)

**Pull request merged**